File: 1-two-sum/two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        my_dict = {}
        
        for each in range(len(nums)):
            if target - nums[each] in my_dict.keys():
                return [each, my_dict[target-nums[each]]]
            else:
                my_dict[nums[each]] = each
        return []

File: 1-two-sum/two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    
    my_dict = {}

    for (let i =0; i<nums.length; i++){
        if (my_dict.hasOwnProperty(target -nums[i])) {
            return [my_dict[target-nums[i]], i];
        }
        else {
            my_dict[nums[i]] = i
        }
    }
};